### DISCUSSION TITLE
Combining two templater code

### DISCUSSION THREAD

**Discussion by Deus333** [SilentVoid13/Templater#1181] [Help]
Is there any way to use <% tp.file.include("[[Template]]") %> this code inside of

<% await tp.system.suggester([ "option 1", "option 2"], [ "<% tp.file.include("[[Template1]]") %>", "<% tp.file.include("[[Template2]]") %>"])%>

like this...
 
I need this for solve this problem https://github.com/zsviczian/obsidian-excalidraw-plugin/issues/1271

**Comment by Zachatoo** [comment]
Here's one way to do that. Set it up in an object so it's easier to manage your templates.

```js
<%*
const options = [
  { label: "option 1", value: "[[Template 1]]" },
  { label: "option 2", value: "[[Template 2]]" },
];
const selectedTemplate = (await tp.system.suggester(option => option.label, options)).value;
tR += await tp.file.include(selectedTemplate);
-%>
```

You could also generate your list of templates by grabbing all files in a folder, see an example of that [here](https://zachyoung.dev/posts/templater-snippets#suggester-for-files-in-a-specific-folder).

**Reply by Zachatoo** [reply]
General rule of thumb: never nest Templater tags.

```
<% stuff <% uh oh, doesn't work %> %>
<% this is fine%> <% so is this %>
```

**Reply by danielo515** [reply]
I think you need  to await the inclusion of the file

**Reply by Zachatoo** [reply]
Updated my code, thanks!
